tests/preprocessing/test_generator.py

```diff
@@ -0,0 +1,133 @@
+"""
+Copyright 2017-2018 Fizyr (https://fizyr.com)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from keras.preprocessing.image import ImageDataGenerator
+from keras_retinanet.preprocessing.generator import Generator
+
+import numpy as np
+import pytest
+
+
+class SimpleGenerator(Generator):
+    def __init__(self, annotations_group):
+        self.annotations_group = annotations_group
+        super(SimpleGenerator, self).__init__(ImageDataGenerator(), group_method='none', shuffle_groups=False)
+
+    def size(self):
+        return len(self.annotations_group)
+
+    def load_annotations(self, image_index):
+        result = self.annotations_group[image_index]
+        return result
+
+
+class TestLoadAnnotationsGroup(object):
+    def test_simple(self):
+        input_annotations_group = [
+            np.array([
+                [  0,   0,  10,  10],
+                [150, 150, 350, 350]
+            ]),
+        ]
+        expected_annotations_group = input_annotations_group
+
+        simple_generator = SimpleGenerator(input_annotations_group)
+        annotations_group = simple_generator.load_annotations_group(simple_generator.groups[0])
+
+        np.testing.assert_equal(expected_annotations_group, annotations_group)
```

Comment:
Normally I prefer to perform the calculation in the assertion itself, since most testing libraries can then print the calculation that failed. I don't know if `np.testing.assert_equal()` can do that though. Probably not, since it is just a function call.

Reply:
numpy shows that there is a x% mismatch between expected and result, and prints the arrays.

Reply:
What I mean is that if you do something like `assert sqrt(4) == 3`, pytest will print `sqrt(4) == 3`, which is more informative when a test fails than `assert expected == actual`.

But in this case I don't think numpy can do the same, so there probably isn't a difference.